src/app/register/page.tsx

```diff
@@ -2,45 +2,65 @@
 import { NextPage } from 'next';
 import { useForm } from 'react-hook-form';
 import { useRouter } from 'next/navigation';
-import React, { useEffect, useState } from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import registerUser from '@/services/user/register';
 import { RegisterUserData, Role, Type } from '@/app/register/types';
 import Form from '@/shared/form/Form';
 import FormInput from '@/shared/formInput/FormInput';
 import FormSelect from '@/shared/formSelect/FormSelect';
 import { Button } from '@/components/ui/button';
+import ReCAPTCHA from 'react-google-recaptcha';
+import { toast, ToastContainer } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
 
 const Register: NextPage = () => {
-  const form = useForm<RegisterUserData & { repeatPassword: string }>();
+  const form = useForm<
+    Omit<RegisterUserData, 'reCaptchaToken'> & { repeatPassword: string }
+  >();
   const watch = form.watch;
   const [role, setRole] = useState<Role | undefined>(Role.USER);
   const [submitting, setSubmitting] = useState(false);
-  const [error, setError] = useState('');
   const router = useRouter();
+  const recaptchaRef = useRef<ReCAPTCHA>(null);
 
   useEffect(() => {
     const subscription = watch((value) => {
       setRole(value.role);
-      setError('');
     });
     return () => subscription.unsubscribe();
   }, [watch]);
 
-  async function submit() {
+  const submit = async () => {
     setSubmitting(true);
+    if (!recaptchaRef?.current?.getValue()) {
+      toast('Molimo vas da potvrdite da niste robot', { type: 'error' });
+      setSubmitting(false);
+      return;
+    }
+
+    const recaptchaValue = recaptchaRef.current.getValue();
     const values = form.getValues();
 
+    setSubmitting(false);
+
     try {
       // eslint-disable-next-line @typescript-eslint/no-unused-vars
       const { repeatPassword, ...rest } = values;
```

Comment:
We can move the setSubmitting into the try statement.
```suggestion
    try {
      setSubmitting(true);
      // eslint-disable-next-line @typescript-eslint/no-unused-vars
      const { repeatPassword, ...rest } = values;
```

Reply:
Yes, I fully agree!